test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -24,49 +25,328 @@ void SignalPhase(const std::wstring& phaseEventName)
+        [](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& /* sender */,
+            PushNotificationCreateChannelStatus const& args)
+        {
+            if (args.status == PushNotificationChannelStatus::InProgress)
+            {
+                // This is basically a noop since it isn't really an error state
+                std::cout << "The channel request is in progress " << args.extendedError << std::endl;
+            }
+            else if (args.status == PushNotificationChannelStatus::InProgressRetry)
+            {
+                std::cout << "The channel request is in back-off retry mode because of a retryable error code: " << args.extendedError << " Expect delays in acquiring it" << std::endl;
+            }
+        });
+
+    // Setup the completed event handler
+    channelOperation.Completed(
+        [&channelEvent](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                auto channelUri = result.Channel().Uri();
+                auto channelExpiry = result.Channel().ExpirationTime();
+
+
+                std::cout << "channelUri: " << winrt::to_string(channelUri.ToString()) << std::endl;
+
+            }
+            else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+            {
+                std::cout << "Failed to complete the async complete handler, gracefully cancel the channelOperation" << std::endl;
+            }
+
+
+            SetEvent(channelEvent.get());
+        });
+
+    // The maximum amount of time it takes for channel request to be obtained - 16mins
+    if (WAIT_OBJECT_0 != WaitForSingleObject(channelEvent.get(), 960000 /* milliseconds */))
+    {
+        std::cout << "Failed to call/handle the async complete handler, gracefully cancel the channelOperation" << std::endl;
+        channelOperation.Cancel();
+        hr = ERROR_TIMEOUT;
     }
-    catch (winrt::hresult_error const& ex)
+    else
     {
-        std::wcout << ex.message().c_str() << std::endl;
+        std::cout << "Channel complete handler has been completed, safely closing the async operation" << std::endl;
+        channelOperation.Close(); // Do not call getresults after this
+    }
+
+    if (hr == S_OK)
+        return true;
+    else
+        return false;
+}
+
+bool MultipleChannelRequestUsingSameRemoteId()
+{
+    wil::unique_handle channelEvent = wil::unique_handle(CreateEvent(nullptr, FALSE, FALSE, nullptr));
+    auto hr = S_OK;
+
+    auto channelOperation1 = PushNotificationManager::CreateChannelAsync(winrt::guid(0xa2e4a323, 0xb518, 0x4799, { 0x9e, 0x80, 0x0b, 0x37, 0xae, 0xb0, 0xd2, 0x25 }));
```

Comment:
I see `channelOperation2` doing something, but I just see this one declared.

If this is your test approach, can you elaborate on the intention here?